routes/tanksRouter.js

```diff
@@ -17,39 +17,41 @@ tanksRouter.route('/')
 
 tanksRouter.route('/:accountID')
   .options(cors.corsWithOptions, (_, res) => { res.sendStatus(200); })
-  .get(cors.cors, async (req, res) => res.json(await getTanksStats(req.params.accountID)))
+  .get(cors.cors, async (req, res) => res.json(await postTanksSnapshots(req.params.accountID)))
   .post(cors.corsWithOptions, authenticate.verifyUser, async (req, res, next) => {
-    const statsToAdd = await postPlayerSnapshots(req.params.accountID);
-    PlayerStats.findOne({ 'data.accountId': req.params.accountID })
-      .then((playerStats) => {
-        if (!playerStats) {
-          PlayerStats.create(statsToAdd)
-            .then((player) => {
-              player.user = req.user._id;
-              player.save();
+    const statsToAdd = await postTanksSnapshots(req.params.accountID);
+    if (!statsToAdd.status) { // if bad data
+      return res.status(404).json(statsToAdd)
+    };
+    statsToAdd.data = statsToAdd.data.filter((tankStats) => tankStats.tank_id);
+    TankStats.findOne({ 'account_id': req.params.accountID })
+      .then((tankStats) => {
+        if (!tankStats) {
+          TankStats.create(statsToAdd)
+            .then((tanks) => {
               res.statusCode = 200;
               res.setHeader('Content-Type', 'application/json');
```

Comment:
The .then() method is replaced with async/await to handle asynchronous operations more cleanly. The try/catch block is used to handle any errors that occur during the asynchronous operations
```suggestion
    try {
      const tankStats = await TankStats.findOne({ 'account_id': req.params.accountID 

      if (!tankStats) {
        const createdTanks = await TankStats.create(statsToAdd);
        res.statusCode = 200;
        res.setHeader('Content-Type', 'application/json');
        return res.json(createdTanks);
    }
```